pkg/webhook/webhook.go

```diff
@@ -102,6 +102,12 @@ func (m *podMutator) Handle(ctx context.Context, req admission.Request) admissio
 	}
 
 	logger := log.Log.WithName("handler").WithValues("pod", pod.Name, "namespace", pod.Namespace, "service-account", serviceAccountName)
+
+	if !pod.GetDeletionTimestamp().IsZero() {
+		logger.Info("pod is being terminated")
+		return admission.Allowed("pod is being terminated")
+	}
+
 	// get service account associated with the pod
 	serviceAccount := &corev1.ServiceAccount{}
 	if err = m.client.Get(ctx, types.NamespacedName{Name: serviceAccountName, Namespace: pod.Namespace}, serviceAccount); err != nil {
```

Comment:
Should we also make the `m.reader.Get` branch below return `admission.Allowed` when the SA is not found?  Or maybe just keep going with some "default" idea of what the SA state is?

Reply:
I think we can do this in `v1.0.0-alpha` release when we are sure every pod request that comes for admission review is intended. Now we rely on the service account to know if this pod needs to be mutated.

Reply:
opened an issue to track this change